src/mobot_client/mobot.py

```diff
@@ -74,277 +82,254 @@ def __init__(self):
+                    customer, customer.phone_number, ChatStrings.COUNTRY_RESTRICTED
+                )
+            else:
+                bst_time = drop_to_advertise.start_time.astimezone(
+                    pytz.timezone(drop_to_advertise.timezone)
+                )
+                response_message = ChatStrings.STORE_CLOSED.format(
+                    date=bst_time.strftime("%A, %b %d"),
+                    time=bst_time.strftime("%-I:%M %p %Z"),
+                    desc=drop_to_advertise.pre_drop_description
+                )
+                self.messenger.log_and_send_message(
+                    customer, customer.phone_number, response_message
                 )
-                transaction_status = receipt_status["receipt_transaction_status"]
-                print("Waiting for", receipt, receipt_status)
 
-            if transaction_status != "TransactionSuccess":
-                print("failed", transaction_status)
-                return "The transaction failed!"
+    def handle_unsolicited_payment(self, customer: Customer, amount_paid_mob: Decimal):
+        self.logger.warning("Could not find drop session for customer; Payment unsolicited!")
+        self.messenger.log_and_send_message(
+            customer, str(customer.phone_number), ChatStrings.UNSOLICITED_PAYMENT
+        )
+        self.payments.send_mob_to_customer(customer, str(customer.phone_number.as_e164), amount_paid_mob, False)
 
-            amount_paid_mob = mc.pmob2mob(receipt_status["txo"]["value_pmob"])
+    # Chat handlers defined in __init__ so they can be registered with the Signal instance
+    def handle_payment(self, source, receipt):
+        self.logger.info(f"Received payment from {source}")
+        receipt_status = None
+        transaction_status = "TransactionPending"
 
-            customer = None
-            drop_session = None
+        if isinstance(source, dict):
+            source = source["number"]
 
-            customer, _ = Customer.objects.get_or_create(phone_number=source)
-            try:
+        self.logger.info("received receipt", receipt)
+        receipt = mc.utility.b64_receipt_to_full_service_receipt(receipt.receipt)
 
-                drop_session = DropSession.objects.get(
-                    customer=customer,
-                    drop__drop_type=DropType.AIRDROP,
-                    state=SessionState.WAITING_FOR_BONUS_TRANSACTION,
-                )
-            except DropSession.DoesNotExist:
-                pass
-            else:
+        while transaction_status == "TransactionPending":
+            receipt_status = self.mcc.check_receiver_receipt_status(
+                self.public_address, receipt
+            )
+            transaction_status = receipt_status["receipt_transaction_status"]
+            self.logger.info(f"Waiting for {receipt}, current status {receipt_status}")
+
+        if transaction_status != "TransactionSuccess":
+            self.logger.error(f"failed {transaction_status}")
```

Comment:
Worth including context on the transaction so that eng could take a look if we ever see this